pkg/rtc/plugins/webm.go

```
@@ -6,33 +6,44 @@ import (
 
 	"github.com/at-wat/ebml-go/webm"
 
+	"github.com/pion/ion/pkg/log"
 	"github.com/pion/rtcp"
 	"github.com/pion/rtp"
 	"github.com/pion/rtp/codecs"
+	"github.com/pion/webrtc/v2"
 	"github.com/pion/webrtc/v2/pkg/media/samplebuilder"
 )
 
 type WebmSaver struct {
+	id                             string
 	audioWriter, videoWriter       webm.BlockWriteCloser
 	audioBuilder, videoBuilder     *samplebuilder.SampleBuilder
 	audioTimestamp, videoTimestamp uint32
 }
 
-func NewWebmSaver() *WebmSaver {
+func NewWebmSaver(id string) *WebmSaver {
 	return &WebmSaver{
+		id:           id,
 		audioBuilder: samplebuilder.New(10, &codecs.OpusPacket{}),
 		videoBuilder: samplebuilder.New(10, &codecs.VP8Packet{}),
 	}
 }
 
 func (s *WebmSaver) ID() string {
-	return "WebMSaver"
+	return s.id
 }
 
 func (s *WebmSaver) Init(...interface{}) {
-
 }
-func (s *WebmSaver) PushRTP(*rtp.Packet) error {
+
+func (s *WebmSaver) PushRTP(pkt *rtp.Packet) error {
+	if pkt.PayloadType == webrtc.DefaultPayloadTypeVP8 {
+		s.PushVP8(pkt)
+		return nil
+	} else if pkt.PayloadType == webrtc.DefaultPayloadTypeOpus {
+		s.PushOpus(pkt)
+		return nil
+	}
```

Should we raise an error here?

---

I think it needs to be returned because WebM does not support encodings other than VP8.

---

Updated to return an error 👍